frontend/pages/sign-in/code.tsx

```diff
@@ -0,0 +1,192 @@
+
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { useRouter } from 'next/router';
+
+import { withLocalizedRequests } from 'hoc/locale';
+
+import { useAppSelector } from 'store/hooks';
+import { session } from 'store/session';
+
+import { InferGetStaticPropsType } from 'next';
+
+import useMe from 'hooks/me';
+
+import { loadI18nMessages } from 'helpers/i18n';
+
+import Alert from 'components/alert';
+import Button from 'components/button';
+import ErrorMessage from 'components/forms/error-message';
+import Input from 'components/forms/input';
+import Loading from 'components/loading';
+import { Paths, UserRoles } from 'enums';
+import NakedLayout, { NakedLayoutProps } from 'layouts/naked';
+import { PageComponent } from 'types';
+import { SignIn, SignInCodeForm } from 'types/sign-in';
+
+import { useSignIn } from 'services/authentication/authService';
+
+export const getStaticProps = withLocalizedRequests(async ({ locale }) => {
+  return {
+    props: {
+      intlMessages: await loadI18nMessages({ locale }),
+    },
+  };
+});
+
+type ProjectDeveloperProps = InferGetStaticPropsType<typeof getStaticProps>;
+
+const CODE_LENGTH = 5;
+
+const SignInCode: PageComponent<ProjectDeveloperProps, NakedLayoutProps> = () => {
+  const { formatMessage } = useIntl();
+  const { replace } = useRouter();
+  const signIn = useSignIn();
+  const { user, isLoading: isUserLoading } = useMe();
+
+  // Email and password got from login page
+  const sessionData = useAppSelector(session);
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setFocus,
+    setValue,
+  } = useForm<SignInCodeForm>({
+    shouldUseNativeValidation: true,
+    reValidateMode: 'onChange',
+  });
+
+  // Redirect if user is signed in
+  useEffect(() => {
+    // Wait until user and inviteUser are loaded to be able to compare them
+    if (!isUserLoading && !!user) {
+      if (user?.role === UserRoles.Light) {
+        // If the user don't have an account go to the 'choose account type' page
+        replace(Paths.AccountType);
+      } else {
+        // If the user has an account go to the dashboard
+        replace(Paths.Dashboard);
+      }
+    }
+  }, [isUserLoading, replace, user]);
+
+  // Create an array of numbers from 0 to the code's length
+  const code = Array.from({ length: CODE_LENGTH }, (_, i) => i);
+
+  useEffect(() => {
+    // Set focus at the first input box
+    setFocus('otp_attempt.0');
+  }, [setFocus]);
+
+  const handleChangeCode = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    const currentInputIndex = Number(name.split('.')[1]);
+    let nextInput = currentInputIndex + 1;
+    if (!value || currentInputIndex === 4) return;
+    setFocus(`otp_attempt.${nextInput}`);
+  };
+
+  const handleSignIn = useCallback(
+    (data: SignIn) => {
+      signIn.mutate(data, {
+        onSuccess: () => {
```

Comment:
On the `pages/signin-in/index.tsx` file, we also have a case where the user accepts an invitation to an account. This is missing from this page. Could you add this? https://github.com/Vizzuality/heco-invest/blob/63111388539ae3c2c818425626933b75dcac1f32/frontend/pages/sign-in/index.tsx#L106-L112